src/main/java/org/project/name/online/book/store/service/shopping/cart/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+package org.project.name.online.book.store.service.shopping.cart.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.project.name.online.book.store.dto.cart.item.CreateCartItemRequestDto;
+import org.project.name.online.book.store.dto.cart.item.UpdateCartItemRequestDto;
+import org.project.name.online.book.store.dto.shopping.cart.ShoppingCartDto;
+import org.project.name.online.book.store.exception.EntityNotFoundException;
+import org.project.name.online.book.store.mapper.cart.item.CartItemMapper;
+import org.project.name.online.book.store.mapper.shopping.cart.ShoppingCartMapper;
+import org.project.name.online.book.store.model.Book;
+import org.project.name.online.book.store.model.CartItem;
+import org.project.name.online.book.store.model.ShoppingCart;
+import org.project.name.online.book.store.model.User;
+import org.project.name.online.book.store.repository.book.BookRepository;
+import org.project.name.online.book.store.repository.cart.item.CartItemRepository;
+import org.project.name.online.book.store.repository.shopping.cart.ShoppingCartRepository;
+import org.project.name.online.book.store.repository.user.UserRepository;
+import org.project.name.online.book.store.service.shopping.cart.ShoppingCartService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public void save(String email, CreateCartItemRequestDto cartItemDto) {
+        User user = userRepository.findByEmail(email).orElseThrow(
+                () -> new EntityNotFoundException("There is no user by email: " + email));
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId()).orElseThrow(
+                () -> new EntityNotFoundException(
+                        "There is no shopping cart by user id: " + user.getId()));
```

Comment:
You have such lines in every public method.
Extract to private method that will provide shoppingCart by email and reuse it

Reply:
I agree. Thanks!